src/app/utils/accountPageUtils.js

```diff
@@ -22,57 +24,162 @@ const makeRequest = (
+          updateAccountHtml,
+          updateErrorMessage,
+          patron.id,
+          buildReqBody(content, { [input.name]: input.value }, locationData),
+          content,
+          toggleLoadingState,
+        );
+      });
+      buttons.push(button);
+      removeTd(input);
+    });
+    buttons.forEach((button) => {
+      const td = document.createElement('td');
+      td.appendChild(button);
+      el.appendChild(td);
+    });
+  });
+  accountPageContent.querySelectorAll('.patFuncRating').forEach(el => el.remove());
+
+  const errorMessageEls = accountPageContent.getElementsByClassName('errormessage');
+  if (errorMessageEls.length) {
+    // in original HTML this is `hidden`
+    errorMessageEls[0].style.display = 'block';
+  }
+
   const eventListenerCb = body => makeRequest(
     updateAccountHtml,
     updateErrorMessage,
     patron.id,
     body,
     content,
   );
+
+  // there are two "Renew All" buttons
+  // we just want the first one
+  let isFirstRenewAll = true;
+  const links = accountPageContent.querySelectorAll('a');
   links.forEach((link) => {
     switch (link.textContent) {
       case 'Sort by Checkout':
-        link.addEventListener('click', () => eventListenerCb({ sortByCheckoutDate: 'bycheckoutdate' }));
+        link.remove();
         break;
       case 'Sort by Due Date':
-        link.addEventListener('click', () => eventListenerCb({ sortByDueDate: 'byduedate' }));
+        link.remove();
+        break;
+      case 'Cancel All':
+        link.remove();
         break;
       case 'Renew All':
-        link.addEventListener('click', () => eventListenerCb({ renewall: 'YES' }));
+        if (isFirstRenewAll) {
+          isFirstRenewAll = false;
+          const button = document.createElement('button');
+          button.textContent = 'Renew All';
+          button.addEventListener('click', (e) => {
+            e.preventDefault();
+            eventListenerCb({ renewall: 'YES' });
+          });
+          button.className = 'button button--filled';
+          link.parentElement.replaceChild(button, link);
+        } else {
+          link.remove();
+        }
         break;
       case 'Renew Marked':
-        link.addEventListener('click', () => eventListenerCb({ renewsome: 'YES', ...selectedItems }));
+        link.remove();
         break;
       default:
-        if (link.href && link.href.includes('/mylists?listNum=')) {
-          link.addEventListener('click', (e) => {
-            e.preventDefault();
-            const splitLink = link.href.split('/mylists?listNum=') || [];
-            const listNum = splitLink[1];
-            eventListenerCb({ listNum });
-          });
-          break;
-        }
         if (link.textContent.includes('Pay Online')) {
           link.href = `https://ilsstaff.nypl.org/webapp/iii/ecom/pay.do?lang=eng&scope=1&ptype=${patron.patronType}&tty=800`;
           link.target = '_blank';
           break;
         }
     }
   });
+  toggleLoadingState(false);
```

Comment:
This should not go here. It should go in `makeRequest` (see that function for more info).

Reply:
The loading state also needs to change after all the DOM manipulation is complete and `makeRequest` is not called. Maybe it could move to `useEffect` after `manipulateAccountPage` Adding the second argument to `useEffect` has fixed the main problem.